v0/destinations/one_signal/util.js

```diff
@@ -0,0 +1,82 @@
+const {
+  getIntegrationsObj,
+  getFieldValueFromMessage,
+  CustomError
+} = require("../../util");
+
+// For mapping device_type value
+const deviceTypeMapping = {
+  android: 1,
+  ios: 0,
+  chrome: 5,
+  safari: 7,
+  firefox: 8
+};
+
+// This function is used to check for the valid device_type value. device_type value should be integer
+// and can be from [0 to 14] only.
+const validateDeviceType = deviceType => {
+  if (Number.isNaN(deviceType) || deviceType < 0 || deviceType > 14) {
+    return false;
+  }
+  return true;
+};
+
+/**
+ * This funnction is used to populate the tags using the traits
+ * @param {} message
+ */
+const populateTags = message => {
+  const tags = {};
+  const traits = getFieldValueFromMessage(message, "traits");
+  const traitsKey = Object.keys(traits);
+  traitsKey.forEach(key => {
+    if (typeof traits[key] === "string") {
+      tags[key] = traits[key];
+    }
+  });
+  if (message.anonymousId) {
+    tags.anonymousId = message.anonymousId;
+  }
+  return tags;
+};
+
+/**
+ * This function is used to populate device_type required for creating a device
+ * @param {*} message
+ * @param {*} payload
+ */
+const populateDeviceType = (message, payload) => {
+  const integrationsObj = getIntegrationsObj(message, "one_signal");
+  const devicePayload = payload;
+  if (integrationsObj && integrationsObj.deviceType) {
```

Comment:
Add check for integrationsObj.identifier as well.

Reply:
updated

Reply:
Done